transmart/api/auth.py

```diff
@@ -6,19 +6,15 @@
 
 class Authenticator(metaclass=abc.ABCMeta):
 
-    def __init__(self, url, username=None, password=None, realm=None):
+    def __init__(self, url, username=None, password=None, realm=None, client_id=None):
         self.url = url
         self.user = username or input('Username: ')
         self.password = password
         self.realm = realm
+        self.client_id = client_id
         self._access_token = None
         self.get_token()
 
-    @property
-    @abc.abstractmethod
-    def client_id(self):
```

Comment:
could change this to `_default_client_id` and update the initialiser to use that if the client_id is None, eg.: `self.client_id = client_id or self._default_client_id`

Reply:
@jochemb I have changed, take a look.

Reply:
lgtm